core/src/com/unciv/logic/automation/unit/BattleHelper.kt

```diff
@@ -52,64 +54,101 @@ object BattleHelper {
+        
+        // We'll probably die next turn if we attack the city
+        if (attacker.baseUnit.isMelee() && attacker.health - BattleDamage.calculateDamageToAttacker(attackerUnit, cityUnit) * 2 <= 0)
+            return 0
+
+        var attackValue = 120 
+        // Siege units should really only attack the city
+        if (attacker.baseUnit.isProbablySiegeUnit()) attackValue += 300
+        // Ranged units don't take damage from the city
+        else if (attacker.baseUnit.isRanged()) attackValue += 50
+        // Lower health cities have a higher priority to attack
+        attackValue -= (city.health - 50) / 2
+
+        // Add value based on number of units around the city
+        val defendingCityCiv = city.civ
+        city.getCenterTile().neighbors.forEach {
+            if (it.militaryUnit != null) {
+                if (it.militaryUnit?.civ == defendingCityCiv)
+                    attackValue -= 10
+                if (it.militaryUnit?.civ == attacker.civ)
+                    attackValue += 20
+            }
         }
+        
+        return attackValue
+    }
 
-        // We're ranged, prioritize that we can kill
-        return unitsToConsider.minBy {
-            Battle.getMapCombatantOfTile(it.tileToAttack)!!.getHealth()
+    /**
+     * Returns a value which represents the attacker's motivation to attack a unit.
+     */
+    private fun getUnitAttackValue(attacker: MapUnit, attackTile: AttackableTile): Int {
+        // Base attack value, there is nothing there...
+        var attackValue = Int.MIN_VALUE
+        // Prioritize attacking military
+        val militaryUnit = attackTile.tileToAttack.militaryUnit
+        if (militaryUnit != null) {
+            attackValue = 0
+            attackValue += 150
+            // Associate enemy units with number of hits from this unit to kill them
+            val attacksToKill = (militaryUnit.health.toFloat() / 
+                BattleDamage.calculateDamageToDefender(MapUnitCombatant(attacker), MapUnitCombatant(militaryUnit))).coerceAtLeast(1f)
+            // We can kill them in this turn
+            if (attacksToKill <= 1) attackValue += 100
+            // On average, this should take around 3 turns, so -30
+            else attackValue -= (attacksToKill * 10).toInt()
+        } else {
+            val civilianUnit = attackTile.tileToAttack.civilianUnit
+            if (civilianUnit != null) {
+                attackValue = 0
+                attackValue += 50
+                // Only melee units should really attack/capture civilian units, ranged units take more than one turn
+                if (attacker.baseUnit.isMelee()) {
+                    if (civilianUnit.isGreatPerson()) {
+                        attackValue += 50
+                        // This is really good if we can kill a great general
+                        if (civilianUnit.isGreatPersonOfType("Great General")) attackValue += 150
```

Comment:
In general, all great people should be prioritized

Reply:
Do you mean 'UniqueType.GreatGeneralProvidesDoubleCombatBonus'? It doesn't sound right though, other than that, I can't find any other Unique to match it.